client/signup/main.jsx

```diff
@@ -161,6 +165,23 @@ class Signup extends React.Component {
 			providedDependencies = pick( queryObject, flow.providesDependenciesInQuery );
 		}
 
+		const searchParams = new URLSearchParams( window.location.search );
+		const isAddNewSiteFlow = searchParams.has( 'ref' );
```

Comment:
I am not sure I understand this. Could you help me understand why we can rely on this?

Reply:
Clicking on "Add new site" from Calypso will take to the signup flow with a `ref` query parameter([code ref](https://github.com/Automattic/wp-calypso/blob/a2db298e5e1f43fa039fa1ec6a8667bb02a64099/client/my-sites/sidebar/index.jsx#L863)). If the ref exists, then the user clicked on "Add new site", and in this case we will delete the signup destination cookie since the user is already inside Calypso.